internal/api/databases.go

```diff
@@ -53,3 +56,40 @@ func handleGetDatabases(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	outputJSON(c, w, databases)
 }
+
+// handleUpdateDatabase responds to PUT /api/database/{database}, updating the
+// database configuration values.
+func handleUpdateDatabase(c *Context, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	databaseID := vars["database"]
+	c.Logger = c.Logger.WithField("database", databaseID)
+
+	database, status, unlockOnce := lockDatabase(c, databaseID)
+	if status != 0 {
+		w.WriteHeader(status)
+		return
+	}
+	defer unlockOnce()
+
+	patchDatabaseRequest, err := model.NewPatchDatabaseRequestFromReader(r.Body)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to decode request")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if patchDatabaseRequest.Apply(database) {
+		err = c.Store.UpdateMultitenantDatabase(database)
+		if err != nil {
+			c.Logger.WithError(err).Error("failed to update database")
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+	}
+
+	unlockOnce()
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusAccepted)
```

Comment:
Should it return `202` or just `200`? I do not see any async process that we start, so I would be in favor of 200.

Reply:
Yep, I think that makes more sense as well. I was mimicking the UpdateCluster endpoint which also has no async processes, but returns the same status. We should probably standardize that one at a later time if we can do so safely.